File: grooby-test-study/src/main/java/com/example/demo/repository/BookRepository.java
package com.example.demo.repository;

import com.example.demo.entity.Book;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface BookRepository extends JpaRepository<Book, Long> {
    Optional<Book> findBookByIsbn(@NonNull String isbn);
}

File: library-app/src/main/kotlin/com/example/libraryapp/domain/book/BookRepository.kt
package com.example.libraryapp.domain.book

import org.springframework.data.jpa.repository.JpaRepository

interface BookRepository : JpaRepository<Book, Long> {

    fun findByName(bookName: String): Book?

//    @Query("select new com.example.libraryapp.dto.book.response.BookStatResponse(b.type, count(b.id)) from Book b group by b.type")
//    fun getStats(): List<BookStatResponse>
}